src/pages/Profile.tsx

```diff
@@ -1,10 +1,254 @@
 import { Link } from 'react-router-dom';
+import Spinner from '../components/Spinner';
+import UserInputString from '../components/UserInputString';
+import { useEffect, useState } from 'react';
+import {
+  AddressForProfile,
+  UserInput,
+  makeAddressesForProfile,
+} from '../utils';
+import { BsInfoCircle } from 'react-icons/bs';
+import { PiPasswordBold } from 'react-icons/pi';
+import { RiLockPasswordLine } from 'react-icons/ri';
+import {
+  CiCirclePlus,
+  CiDeliveryTruck,
+  CiEdit,
+  CiFileOn,
+  CiTrash,
+} from 'react-icons/ci';
+import { CiBookmarkCheck } from 'react-icons/ci';
+import { useApiGetCustomer } from '../hooks';
+import { Customer } from '@commercetools/platform-sdk';
 
+type AddressProps = { address: AddressForProfile };
+function Address({ address }: AddressProps) {
+  return (
+    <>
+      <div className="flex flex-row justify-end gap-1 text-2xl">
+        {address.isDefault && <CiBookmarkCheck title="Default" />}
+        {address.isDefault && address.isShipping && (
+          <CiDeliveryTruck title="Shipping Address" />
+        )}
+        {address.isDefault && address.isBilling && (
+          <CiFileOn title="Shipping Address" />
+        )}
+        {!address.isDefault && address.isShipping && (
+          <CiDeliveryTruck title="Shipping Address" />
+        )}
+        {!address.isDefault && address.isBilling && (
+          <CiDeliveryTruck title="Shipping Address" />
+        )}
+      </div>
+      <p>{address.strAddress}</p>
+      <div className="flex flex-row gap-4 text-2xl">
+        <a href="" title="Edit" className=" hover:text-moonNeutral-600">
+          <CiEdit />
+        </a>
+        {!address.isDefault && (
+          <a href="" title="Delete" className=" hover:text-moonNeutral-600">
+            <CiTrash />
+          </a>
+        )}
+      </div>
+    </>
+  );
+}
 function Profile() {
+  const { loading: isLoading, ok, customer } = useApiGetCustomer();
+  const isUpdateProfile = false;
+  const isUpdateAddress = false;
+
+  const [editProfile, setEditProfile] = useState(false);
+
+  const [firstName, setFirstName] = useState('');
+  const [validFirstName, setValidFirstName] = useState(false);
+  const [cluesVisibleFirstName, setCluesVisibleFirstName] = useState(false);
+  const [firstNameFocus, setFirstNameFocus] = useState(false);
```

Comment:
все эти `useState` можно преобразовать в один, который будет хранить объект

Reply:
Мне сложно. Можно ссылку поясняющую?   
Если ты про [Should I use one or many state variables?](https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables) Возможно это устарело. Но как я, понял отдельные useState лучше - не запуташься. Для меня сейчас это важнее :)

Reply:
Ладно, уговорил. Сделал один обект

Reply:
Я почитал, да, там есть момент, что если в объекте будут несвязанные свойства, то лучше отдельные состояния, но если сделал, то тоже хорошо

Reply:
Поторопиося я.
```
  useEffect(() => {
    const valid = UserInput.checkPasswordRequiredValid(pwd.value);
    setPwd({
      ...pwd,
      valid,
      visibleClue: pwd.focus && (!pwd.value || !valid),
    });
  }, [pwd]);
```
раньше у меня было несколько состояний и такая конструкция работала. А теперь состояние одно и поcтоянно меняется и как огрничить я не понимаю
>Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.

Reply:
Разобрался.
```useEffect(() => {
    const valid = UserInput.checkPasswordRequiredValid(pwd.value);
    setPwd((prevState) => ({
      ...prevState,
      valid,
      visibleClue: pwd.focus && (!pwd.value || !valid),
    }));
  }, [pwd.value, pwd.focus]);```

Reply:
Объектом может удобно быть когда много, ведь это не очень оптимально, так как надо создавать новый объект и им перезаписывать и потом браузеру надо удалить старый объект. Я имею что не всегда лучше его использовать.